runtime/platform.c

```diff
@@ -154,10 +157,15 @@ static struct lf_skiplist mmap_blocks = {NULL};
 
 void* caml_mem_map(uintnat size, uintnat alignment, int reserve_only)
 {
-  uintnat alloc_sz = caml_mem_round_up_pages(size + alignment);
+  CAMLassert(Is_power_of_2(alignment));
+  alignment = caml_mem_round_up_pages(alignment);
+
+  uintnat alloc_sz = size + alignment;
   void* mem;
   uintnat base, aligned_start, aligned_end;
 
+  CAML_ASSERT_SIZE_IS_PAGE_ALIGNED(size);
```

Comment:
Why not `CAMLassert(Is_page_aligned(size))`, following the Is_power_of_2 style above?

Reply:
Much better that way, yes 🤦